Mathlib/Analysis/SpecialFunctions/ContinuousFunctionalCalculus/Rpow/Basic.lean

```diff
@@ -322,6 +322,32 @@ lemma sqrt_map_pi {c : ∀ i, C i} (hc : ∀ i, 0 ≤ c i := by cfc_tac) :
 
 end pi
 
+/-- For an element `a` in a C⋆-algebra, TFAE:
+* `0 ≤ a`
+* `a = sqrt a * sqrt a`
+* `a = b * b` for some non-negative `b`
+* `a = b * b` for some self-adjoint `b`
+* `a = star b * b` for some `b`
+* `a = b * star b` for some `b`
+* `a` is self-adjoint and has a non-negative spectrum -/
+theorem _root_.nonneg_TFAE {a : A} :
+    [ 0 ≤ a,
+      a = sqrt a * sqrt a,
+      ∃ b : A, 0 ≤ b ∧ a = b * b,
+      ∃ b : A, IsSelfAdjoint b ∧ a = b * b,
+      ∃ b : A, a = star b * b,
+      ∃ b : A, a = b * star b,
+      IsSelfAdjoint a ∧ QuasispectrumRestricts a ContinuousMap.realToNNReal ].TFAE := by
+  tfae_have 1 ↔ 7 := nonneg_iff_isSelfAdjoint_and_quasispectrumRestricts
+  tfae_have 5 → 1 := fun ⟨b, hb⟩ => hb ▸ star_mul_self_nonneg _
```

Comment:
If you do `5 \to 6` (easy) instead of `4 \to 6` then this is redundant.

And can you write them in order in the proof (except for 7)?